symengine/polynomial.h

```diff
@@ -244,17 +244,44 @@ class UnivariateExprPolynomial
     {
     }
     UnivariateExprPolynomial(int i)
-        : poly_(UnivariatePolynomial::create(symbol(""), {{0, Expression(i)}}))
+        : poly_(UnivariatePolynomial::create(symbol(""), {{Expression(i)}}))
+    {
+    }
+    UnivariateExprPolynomial(std::string varname) 
+        : poly_(UnivariatePolynomial::create(symbol(varname), {0 ,1}))
     {
     }
     UnivariateExprPolynomial(RCP<const UnivariatePolynomial> p)
         : poly_(std::move(p))
     {
     }
     UnivariateExprPolynomial(Expression expr)
-        : poly_(UnivariatePolynomial::create(symbol(""), {{0, expr}}))
     {
+        if (is_a<Pow>(*expr.get_basic())) {
```

Comment:
`UnivariateExprPolynomial` wasn't building polynomials like `x^(-1)` correctly from `div(one, x)` without this code. This made doing operations like multiplication come out with the wrong answer.